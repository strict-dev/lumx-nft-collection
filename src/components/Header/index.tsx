import Logo from '@public/assets/logo-seamore.svg'
import MenuIcon from '@public/assets/menu.svg'
import Search from 'components/Search'
import Image from 'next/image'

const Header = () => {
  return (
    <header className="fixed z-50 flex items-center justify-between w-full max-w-sm p-4 mb-6 border-b h-14 bg-neutral-800 border-b-neutral-700">
      <div className="flex items-center justify-between gap-4">
        <Image src={Logo} alt="Logo" width={24} height={24} layout="fixed" />
        <div className="flex gap-1">
          <span>71</span>
          <span className="font-bold text-neutral-600">GWEI</span>
        </div>
      </div>

      <nav className="flex items-center justify-between gap-4">
        <Search />

        <Image
          src={MenuIcon}
          alt="Menu"
          width={24}
          height={24}
          layout="fixed"
          onClick={() => console.log('Menu')}
          className="cursor-pointer"
        />
      </nav>
    </header>
  )
}

export default Header
