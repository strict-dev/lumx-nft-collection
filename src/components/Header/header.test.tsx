import { render } from '@testing-library/react'

import Header from '.'

describe('<Header />', () => {
  it('should render the heading', () => {
    const { container } = render(<Header />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
