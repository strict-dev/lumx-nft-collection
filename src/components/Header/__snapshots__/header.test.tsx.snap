// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render the heading 1`] = `
<header
  class="fixed z-50 flex items-center justify-between w-full max-w-sm p-4 mb-6 border-b h-14 bg-neutral-800 border-b-neutral-700"
>
  <div
    class="flex items-center justify-between gap-4"
  >
    <span
      style="box-sizing: border-box; display: inline-block; overflow: hidden; width: 24px; height: 24px; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
    >
      <img
        alt="Logo"
        data-nimg="fixed"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
      />
      <noscript />
    </span>
    <div
      class="flex gap-1"
    >
      <span>
        71
      </span>
      <span
        class="font-bold text-neutral-600"
      >
        GWEI
      </span>
    </div>
  </div>
  <nav
    class="flex items-center justify-between gap-4"
  >
    <button
      class="flex items-center px-3 py-1 rounded-sm transition
      
      focus-within:outline focus-within:outline-neutral-700"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; width: 24px; height: 24px; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
      >
        <img
          alt="Search"
          data-nimg="fixed"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
      <input
        class="px-3 bg-transparent w-36 outline-none transition : ;
        hidden"
        placeholder="Search"
        type="text"
      />
    </button>
    <span
      style="box-sizing: border-box; display: inline-block; overflow: hidden; width: 24px; height: 24px; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
    >
      <img
        alt="Menu"
        class="cursor-pointer"
        data-nimg="fixed"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
      />
      <noscript />
    </span>
  </nav>
</header>
`;
